test/if.civet

```diff
@@ -282,9 +282,7 @@ describe "if", ->
     ---
     x if { x } := y
     ---
-    let ref;if (ref = y) {
-    const { x } = ref;x
-    }
+    let ref;if ((ref = y) && 'x' in ref) { const { x } = ref;x }
```

Comment:
Whoa, this is pretty interesting. But I think makes sense; `:=` is finally becoming a pattern-matching operator.